PracticeProject/ViewModel/UserViewModel.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  UserViewModel.swift
+//  PracticeProject
+//
+//  Created by Gnanavel on 31/03/22.
+//
+
+import Foundation
+import Combine
+
+
+/// UserViewModel is handling the user selection functionality like api and other business logics
+class UserViewModel: ObservableObject {
+    @Published var userList = [UserModel]()
```

Comment:
What is the purpose of `@Published` and elaborate how it works with SwiftUI views

Reply:
`@Published` is a property wrapper in SwiftUI, I have created the ViewModel as `ObservableObject`. So, SwiftUI views will automatically monitor the changes of the ViewModel property wrapper. If we made any changes to this object, It will automatically observe and update the latest data to views.

Reply:
Do you know the run-time cost of this observation operations? Does it affect performance? Are there different options of implementation to achieve the same goal? 
(Try to discuss what you know and have read about this. The question is not to say this is wrong but to open up this discussion)